### PR TITLE
feat: pay to batcher with keystore

### PR DESCRIPTION
**Pull request by NicolasRampoldi**
# Description

- This PR allows the client to pay to the batcher.

# To Test

- Import/Create Keystore file, you can follow README_SEND_PROOFS.md instructions for this
- Make sure you have the latest version of the aligned contracts deployed.
- Run `make install_aligned_compiling`
- Run `make anvil_start`.
- Fund your wallet:

```
cast send <address> --value 1ether --rpc-url http://localhost:8545/ --private-key 0x2a871d0798f97d79848a013d4936a73bf4cc922c825d33c1cf7073dff6d409c6
```

- Send to the batcher:

```
aligned deposit-to-batcher --keystore_path <keystore_path> --amount 0.1ether
```

- Check your balance in the batcher:

```
cast call 0x7969c5eD335650692Bc04293B07F5BF2e7A673C0 "UserBalances(address)(uint256)" <address>
```

It should be **100000000000000000**.

Depends on #460.

### CODE REVIEW

File: batcher/aligned/src/main.rs [hunk 1]
```diff
@@ -59,6 +61,12 @@ pub enum AlignedCommands {
         name = "get-vk-commitment"
     )]
     GetVerificationKeyCommitment(GetVerificationKeyCommitmentArgs),
+    // GetVericiationKey, command name is get-vk-commitment
+    #[clap(
+        about = "Deposits Ethereum in the batcher to pay for proofs",
```
Comment (uri-99) [inline batcher/aligned/src/main.rs:53]:
```suggestion
        about = "Deposit Ethereum in the Batcher Payment Service to fund the verification of your proofs",
```

File: batcher/aligned/src/main.rs [hunk 2]
```diff
@@ -102,6 +110,37 @@ pub struct SubmitArgs {
     keystore_path: Option<PathBuf>,
 }
 
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+pub struct DepositToBatcherArgs {
+    #[arg(
+        name = "Batcher Eth Address",
+        long = "conn",
```
Comment (uri-99) [inline batcher/aligned/src/main.rs]:
```suggestion
        long = "payment-service-address",
```

File: batcher/aligned/src/main.rs [hunk 3]
```diff
@@ -102,6 +110,37 @@ pub struct SubmitArgs {
...
+        default_value = "0x7969c5eD335650692Bc04293B07F5BF2e7A673C0"
+    )]
+    batcher_eth_address: String,
+    #[arg(
+        name = "Path to local keystore",
+        long = "keystore_path",
```
Comment (uri-99) [inline batcher/aligned/src/main.rs:113]: the other commands/flags seem to be kebab-case
```suggestion
        long = "keystore-path",
```

File: batcher/aligned/src/main.rs [hunk 4]
```diff
@@ -102,6 +110,37 @@ pub struct SubmitArgs {
...
+        required = true
+    )]
+    keystore_path: Option<PathBuf>,
+    #[arg(
+        name = "Ethereum RPC provider address",
+        long = "rpc",
```
Comment (uri-99) [inline batcher/aligned/src/main.rs:119]: should we match `cast --rpc-url` flag format?
```suggestion
        long = "rpc-url",
```

File: batcher/aligned/src/main.rs [hunk 5]
```diff
@@ -325,6 +364,88 @@ async fn main() -> Result<(), errors::BatcherClientError> {
...
+            let client = SignerMiddleware::new(eth_rpc_provider.clone(), wallet.clone());
+
+            let balance = client
+                .get_balance(wallet.address(), None)
+                .await
+                .map_err(|e| {
+                    BatcherClientError::EthError(format!("Error while getting balance: {}", e))
+                })?;
+
+            if balance <= U256::from(0) {
```
Comment (uri-99) [inline batcher/aligned/src/main.rs]: should we check it against the amount user want to send?
```suggestion
            if balance <= U256::from(amount) {
```

**Review by uri-99 — CHANGES_REQUESTED** [review]

File: batcher/aligned/src/main.rs (hunk 4 above)
Comment (NicolasRampoldi) [reply batcher/aligned/src/main.rs:119]: It is the same convention in the other commands, I agree tho

File: batcher/aligned/src/main.rs (hunk 2 above)
Comment (NicolasRampoldi) [reply batcher/aligned/src/main.rs]: It is the same convention in the other commands, I agree tho

File: batcher/aligned/src/main.rs (hunk 3 above)
Comment (NicolasRampoldi) [reply batcher/aligned/src/main.rs:113]: It is the same convetion in the other commands, I agree tho

File: batcher/aligned/src/main.rs (hunk 5 above)
Comment (NicolasRampoldi) [reply batcher/aligned/src/main.rs]: Solved in [9c0b9b5](https://github.com/yetanotherco/aligned_layer/pull/514/commits/9c0b9b595ad463258932a545b4db07c9cf0b0810).

**Review by uri-99 — APPROVED** [review]

File: README_SEND_PROOFS.md [hunk 6]
```diff
@@ -35,7 +35,30 @@ This will create the ECDSA keystore file in `~/.aligned_keystore/keystore0`
...
+aligned deposit-to-batcher --keystore_path <keystore_path> --amount 0.1ether
+```
+
+This commands also allows the usage of the flags: 
+- `--batcher_addr` to specify the address of the Batcher Payment Service smart contract.
+- `--rpc` to specify the rpc url to be used.
+- `--chain` to specify the chain id to be used.
+- Note: `--amount` flag parameter must be with the shown format, followed by the `ether` keyword to specify how many ethers you wish to deposit to the batcher.
+
+After depositing funds, you can verify the Service has correctly recieved them, executing the following command:
```
Comment (entropidelic) [inline README_SEND_PROOFS.md]:
```suggestion
After depositing funds, you can verify the Service has correctly received them, executing the following command:
```

**Review by taturosati — APPROVED** [review]
lgtm

**Review by entropidelic — APPROVED** [review]

**Pull request merged**